tests/test_client.py

```diff
@@ -0,0 +1,159 @@
+                                         'EVENT_RESPONSES',
+                                         'IMPRESSIONS',
+                                         'LEAD_GENERATION',
+                                         'QUALITY_LEAD',
+                                         'LINK_CLICKS',
+                                         'OFFSITE_CONVERSIONS',
+                                         'PAGE_LIKES',
+                                         'POST_ENGAGEMENT',
+                                         'QUALITY_CALL',
+                                         'REACH',
+                                         'LANDING_PAGE_VIEWS',
+                                         'VISIT_INSTAGRAM_PROFILE',
+                                         'VALUE',
+                                         'THRUPLAY',
+                                         'DERIVED_EVENTS',
+                                         'APP_INSTALLS_AND_OFFSITE_CONVERSIONS',
+                                         'CONVERSATIONS',
+                                         'IN_APP_VALUE',
+                                         'MESSAGING_PURCHASE_CONVERSION',
+                                         'MESSAGING_APPOINTMENT_CONVERSION',
+                                         'SUBSCRIBERS',
+                                         'REMINDERS_SET']
+
+        # valid and verified objectives listed below, other objectives should be re-mapped to these
+        self.campaign_objectives = ['OUTCOME_APP_PROMOTION',
+                                    'OUTCOME_AWARENESS',
+                                    'OUTCOME_ENGAGEMENT',
+                                    'OUTCOME_LEADS',
+                                    'OUTCOME_SALES',
+                                    'OUTCOME_TRAFFIC']
+
+    def get_account_objects(self, stream):
+        assert stream in  self.stream_endpoint_map.keys(), \
+            f'Endpoint undefined for specified stream: {stream}'
+        endpoint = self.stream_endpoint_map[stream]
+        url = self.account_url + endpoint
+        params = {'access_token': self.access_token,
+                  'limit': 100}
+        LOGGER.info(f"Getting url: {url}")
+        response = requests.get(url, params)
+        response.raise_for_status()
+        LOGGER.info(f"Returning get response: {response}")
+        return response.json()
+
+    def create_account_objects(self, stream):
+        assert stream in  self.stream_endpoint_map.keys(), \
+            f'Endpoint undefined for specified stream: {stream}'
+        endpoint = self.stream_endpoint_map[stream]
+        url = self.account_url + endpoint
+        LOGGER.info(f"Posting to url: {url}")
+        params = self.generate_post_params(stream)
+        response = requests.post(url, params)
+        response.raise_for_status()
+        LOGGER.info(f"Returning post response: {response}")
+        return response
+
+    def generate_post_params(self, stream):
```

Comment:
This is more than enough for now if we have random pk values.  For an all fields test we would need to expand this and it could get messy.  Might need to think about that more in the future, but not necessary now.

Also, to think about for the future (and maybe for the pagination test) is if we should store and return the data that we get and create to compare to what the sync gets and to make sure it is getting the correct data.  Not sure if our standard pagination test worries about that but probably not since the data is unknown ahead of time.

Reply:
Noted.